lib/IDE/Refactoring.cpp

```diff
@@ -5397,6 +5404,149 @@ class AsyncConverter : private SourceEntityWalker {
     return true;
   }
 
+  /// Creates an async alternative function that forwards onto the completion
+  /// handler function through
+  /// withCheckedContinuation/withCheckedThrowingContinuation.
+  bool createAsyncWrapper() {
+    assert(Buffer.empty() && "AsyncConverter can only be used once");
+    auto *FD = cast<FuncDecl>(StartNode.get<Decl *>());
+
+    // First add the new async function declaration.
+    addFuncDecl(FD);
+    OS << tok::l_brace << "\n";
+
+    // Then add the body.
+    OS << tok::kw_return << " ";
+    if (TopHandler.HasError)
+      OS << tok::kw_try << " ";
+
+    OS << "await ";
+
+    // withChecked[Throwing]Continuation { cont in
+    if (TopHandler.HasError) {
+      OS << "withCheckedThrowingContinuation";
+    } else {
+      OS << "withCheckedContinuation";
+    }
+    OS << " " << tok::l_brace << " cont " << tok::kw_in << "\n";
+
+    // asyncFn(args...) {
+    addCallToAsyncMethod(FD, TopHandler);
+    OS << " " << tok::l_brace;
+
+    auto SuccessParams = TopHandler.getSuccessParams();
+
+    // Prepare names for the main trailing closure, as well as reference
+    // strings that may include e.g force unwraps to access the value.
+    SmallVector<SmallString<4>, 2> SuccessParamNames;
+    SmallVector<SmallString<8>, 2> SuccessParamRefs;
+    for (auto idx : indices(SuccessParams)) {
+      auto &Param = SuccessParams[idx];
+      SmallString<8> Name;
```

Comment:
Could just add to paramnames/refs straight away rather than creating and then copying into the vector, but it doesn't really matter.